pkg/asset/machines/worker.go

```diff
@@ -89,114 +87,88 @@ func (w *Worker) Generate(dependencies asset.Parents) error {
 	}
 
 	ic := installconfig.Config
-	pool := workerPool(ic.Machines)
-	switch ic.Platform.Name() {
-	case awstypes.Name:
-		mpool := defaultAWSMachinePoolPlatform()
-		mpool.InstanceType = "m4.large"
-		mpool.Set(ic.Platform.AWS.DefaultMachinePlatform)
-		mpool.Set(pool.Platform.AWS)
-		if len(mpool.Zones) == 0 {
-			azs, err := aws.AvailabilityZones(ic.Platform.AWS.Region)
-			if err != nil {
-				return errors.Wrap(err, "failed to fetch availability zones")
-			}
-			mpool.Zones = azs
-		}
-		pool.Platform.AWS = &mpool
-		sets, err := aws.MachineSets(clusterID.ClusterID, ic, &pool, string(*rhcosImage), "worker", "worker-user-data")
-		if err != nil {
-			return errors.Wrap(err, "failed to create worker machine objects")
-		}
-
-		list := listFromMachineSets(sets)
-		raw, err := yaml.Marshal(list)
-		if err != nil {
-			return errors.Wrap(err, "failed to marshal")
-		}
-		w.MachineSetRaw = raw
-	case libvirttypes.Name:
-		mpool := defaultLibvirtMachinePoolPlatform()
-		mpool.Set(ic.Platform.Libvirt.DefaultMachinePlatform)
-		mpool.Set(pool.Platform.Libvirt)
-		pool.Platform.Libvirt = &mpool
-		sets, err := libvirt.MachineSets(clusterID.ClusterID, ic, &pool, "worker", "worker-user-data")
-		if err != nil {
-			return errors.Wrap(err, "failed to create worker machine objects")
-		}
 
-		list := listFromMachineSets(sets)
-		raw, err := yaml.Marshal(list)
-		if err != nil {
-			return errors.Wrap(err, "failed to marshal")
-		}
-		w.MachineSetRaw = raw
-	case nonetypes.Name:
-		// This is needed to ensure that roundtrip generate-load tests pass when
-		// comparing this value. Otherwise, generate will use a nil value while
-		// load will use an empty byte slice.
-		w.MachineSetRaw = []byte{}
-	case openstacktypes.Name:
-		mpool := defaultOpenStackMachinePoolPlatform(ic.Platform.OpenStack.FlavorName)
-		mpool.Set(ic.Platform.OpenStack.DefaultMachinePlatform)
-		mpool.Set(pool.Platform.OpenStack)
-		pool.Platform.OpenStack = &mpool
-
-		sets, err := openstack.MachineSets(clusterID.ClusterID, ic, &pool, string(*rhcosImage), "worker", "worker-user-data")
-		if err != nil {
-			return errors.Wrap(err, "failed to create master machine objects")
-		}
+	machineSets := []runtime.Object{}
 
-		list := listFromMachineSetsDeprecated(sets)
-		w.MachineSetRaw, err = yaml.Marshal(list)
-		if err != nil {
-			return errors.Wrap(err, "failed to marshal")
+	for _, pool := range ic.Machines {
+		if pool.Name == validation.MasterPoolName {
+			continue
 		}
-	default:
-		return fmt.Errorf("invalid Platform")
-	}
-	return nil
-}
-
-func workerPool(pools []types.MachinePool) types.MachinePool {
-	for idx, pool := range pools {
-		if pool.Name == "worker" {
-			return pools[idx]
+		switch ic.Platform.Name() {
+		case awstypes.Name:
+			mpool := defaultAWSMachinePoolPlatform()
+			mpool.InstanceType = "m4.large"
```

Comment:
Do we care about prioritizing this PR vs. #1163?  They will conflict on at least this line.

Reply:
I don't care about prioritizing. I can rebase this PR easily enough.